VideoLocker/src/main/java/org/edx/mobile/discussion/DiscussionTextUtils.java

```diff
@@ -50,6 +50,12 @@ public static void setAuthorAttributionText(@NonNull TextView textView,
             final CharSequence formattedTime = getRelativeTimeSpanString(context,
                     initialTimeStampMs, authorData.getCreatedAt().getTime());
             final String authorLabel = authorData.getAuthorLabel();
+            final StringBuilder authorLabelFormatted = new StringBuilder();
+            if (authorLabel != null) {
+                authorLabelFormatted.append("(");
+                authorLabelFormatted.append(authorLabel);
+                authorLabelFormatted.append(")");
```

Comment:
**TL;DR:**
What you are suggesting would be an overkill.

**Longer version**
The parenthesis is a basic construct in any language that I've come across.

But, if we are to localize the string with parenthesis included then, we'll have to added 3 new strings of the following:
````xml
<!-- The date and author of a discussion comment, question, or thread, e.g. "3 days ago by Brian STAFF" -->
<string name="post_attribution">{time} by {author} {author_label}</string>
<!-- The date and author of an answer to a discussion question e.g. "Marked as answer 2 days ago by Khalid TA" -->
<string name="answer_author_attribution">Marked as answer {time} by {author} {author_label}</string>
<!-- The date and author of an endorsed response to a discussion thread e.g. "Endorsed 2 days ago by Khalid TA" -->
<string name="endorser_attribution">Endorsed {time} by {author} {author_label}</string>
````

The strings above will then have parenthesis around `{author_label}` and their counterpart won't have the `{author_label}` altogether for the case when server returns null for `authorLabel`.

Other than that, we'll have to change all the callers of our utility function with the `null` check for `authorLabel` to provide the proper `stringRes`.